x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/FollowIndexAction.java

```diff
@@ -173,7 +189,11 @@ protected void doExecute(Request request, ActionListener<Response> listener) {
             if (remoteClusterIndices.containsKey(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)) {
                 // Following an index in local cluster, so use local cluster state to fetch leader IndexMetaData:
                 IndexMetaData leaderIndexMetadata = localClusterState.getMetaData().index(request.leaderIndex);
-                start(request, null, leaderIndexMetadata, followIndexMetadata, listener);
+                try {
+                    start(request, null, leaderIndexMetadata, followIndexMetadata, listener);
+                } catch (IOException e) {
+                    listener.onFailure(e);
```

Comment:
add a return method here just in case? I don't like this construct but can't think of how to improve it (all other options I see suck too)

Reply:
I don't think it is really needed, because this were the if code block ends and below it is an else code block and that is it.

Reply:
I know it's not needed now but it's a bug waiting to happen when someone adds a line below.

Reply:
got it 👍